### ISSUE TITLE
Query multiple metrics with name pattern

### ISSUE THREAD

**Issue by JozoVilcek** [kairosdb/kairosdb#206]
Is it possible with kairosdb to query and match multiple metrics based on name pattern? E.g. get me all metrics where name = "cpu.*" which would return me "cpu-idle", "cpu-nice", ...

**Comment by FrEaKmAn** [comment]
I don't think so, but you could provide a list of all metrics. I saw the #207 issue. I think solr search plugin resolves/will resolve this problem.

**Comment by JozoVilcek** [comment]
So the plan is to support such feature via search engine. Can you elaborate a bit? I assume one will have to provide a reference to a running Solr (not part of kairosdb). Or? Another moving part is always a bit suboptimal.
I know it is hard question, but do you have any estimate on when this could be ready to use?

**Comment by FrEaKmAn** [comment]
Solr is great for this kind of things, especially when you have a large number of metrics and tags. Yes, I agree that it's another moving part. But on the other side, integrating this into KairosDB can increase the numbers of bugs and problems.

There is a repo you can check out - https://github.com/kairosdb/kairos-solr.

**Comment by KlavsKlavsen** [comment]
To me - coming from graphite.. Its absolutely invalueable being able to give f.ex. tag: host=[p-db*] - or metric range like curl.* (we have a lot of curl metrics - I want all shown on same graph, including new ones that show up rather often).

**Comment by brianhks** [comment]
Yes I do like this feature.  I intent to add it after the cql work is done.

**Comment by brianhks** [comment]
Here is my proposal, not perfect but a proposal.  There would be a configurable prefix that if you put in front of your metric name in a query, kairos will do a prefix search on the metric names and then query all metrics returned.  For example:

Say you have the following metrics:
foo.bar
foo.baz
foo.slim

And my prefix is "prefix:"

So if in my query I put the metric name as "prefix:foo.b", Kairos will do a prefix search on the metric names using "foo.b" and the backend would query foo.bar and foo.baz and return the results as if it was one metric.

That bit wouldn't be very hard.  Future work would be to add a group_by metric name.
